packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -339,3 +341,19 @@ class AlwaysScrollableScrollPhysics extends ScrollPhysics {
   @override
   bool shouldAcceptUserOffset(ScrollMetrics position) => true;
 }
+
+/// Scroll physics that only lets the user scroll if there is content to scroll,
+/// and in general has default behavior.
+///
+/// See also:
+///
+///  * [AlwaysScrollableScrollPhysics], which is similar except that the user
+///    can always attempt to scroll, even when there isn't sufficient content.
+class DefaultScrollPhysics extends ScrollPhysics {
```

Comment:
I'd remove this and make ScrollPhysics play this role.  Also, test that ScrollPhysics works in this capacity.